### ISSUE TITLE
pretrained_bert model

### ISSUE THREAD

**Issue by seun-wun** [whgaara/pytorch-soft-masked-bert#3]
你好 这个项目可以加载bert预训练好的模型吗？比如chinese_L-12_H-768_A-12

**Comment by whgaara** [comment]
你好，是这样的，第一我这是torch版的，你只能用torch的预训练模型，例如哈工大的那款。第二这个模型分成两部分，第一部分的检测阶段应该没有靠谱的预训练模型，但是后面的纠正部分可以直接从哈工大的模型里提取参数，就是transformerblock的参数。当然，如果使用预训练模型就一定要注意vocab.txt文件跟别人的要尽量保持一致了。
如果对你有帮助，麻烦给颗小星星，谢谢啦。
具体的加载预训练模型的方法可以看我另一个roberta的代码，跟这个是一套体系的写法，里面有加载的模型链接和方法。



发自我的iPhone


------------------ Original ------------------

**Comment by lmw0320** [comment]
@whgaara  想请教大佬几点问题：
1. char_meta.txt中貌似有点小错误，比如'重'字，发音有2个，zhong4和chong2，而文件中居然还有个tong2？？ 这个文件是哪来的？
2.  checkpoint中的pretrain文件夹下的vocab.txt，与bert的预训练模型中的vocab.txt一样么？ ---我是下载了pytorch版本的bert预训练模型，是否可以直接用？还是要把其中的vocab.txt替换掉？

**Comment by whgaara** [comment]
font{
        line-height: 1.6;
    }
    ul,ol{
        padding-left: 20px;
        list-style-position: inside;
    }


    
    你好，感谢你的阅读：1.char_meta是我之前从爱奇艺的论文faspell中获取的；2.vocab.txt默认就是用的官方的，数量为21128，不需要更换。
    
        
            
        
    
    
      
    
        
                    
            
                
                    
                            
                                
                            
                            
                                695509060
                            
                    
                        
                            
                                    ***@***.***
                            
                        
                
            
        
        
    
    
        签名由
        网易邮箱大师
        定制
    
 
    

    On 05/13/2021 ***@***.***> wrote： 



@whgaara  想请教大佬几点问题：

char_meta.txt中貌似有点小错误，比如'重'字，发音有2个，zhong4和chong2，而文件中居然还有个tong2？？ 这个文件是哪来的？
checkpoint中的pretrain文件夹下的vocab.txt，与bert的预训练模型中的vocab.txt一样么？ ---我是下载了pytorch版本的bert预训练模型，是否可以直接用？还是要把其中的vocab.txt替换掉？


—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by lmw0320** [comment]
@whgaara 感谢大佬答复。
1. 我后面用了较大规模的训练数据集，发现数据转换的时候，提示有几百个字符不在vocab.txt中，那我要如何处理呢？ 直接加入到vocab.txt中，貌似预训练模型就没法用了??
2.  在step1_gen_train_test.py中的random_wrong，是将原始正确文本进行随机的单个字符替换吧？ 代码number = random.randint(672, 7992)中的672和7992是随便写的？ 我看gen_train_test函数中，train_data没进行处理，而是放在smbert_dataset.py文件中的DataFactory方法中处理，而且该处的处理方法好像也只是随机替换，并没有涉及同音字，形近字等？---也就是说char_meta.txt文件并没有用到？？
3. 如果只是简单的文字替换的话，那原始文本和标签文本的前后长度，还是一样的吧？而实际场景中，错误的句子，很可能是有一定的增减几率的。。而且，更多的情况下的错误会是同音字、谐音字，形近字的错误。。这样的话，模型训练出来，对上述提到的错误情况，是否有用呢？

**Comment by GioGioBond** [comment]
> @whgaara 感谢大佬答复。
> 
> 1. 我后面用了较大规模的训练数据集，发现数据转换的时候，提示有几百个字符不在vocab.txt中，那我要如何处理呢？ 直接加入到vocab.txt中，貌似预训练模型就没法用了??
> 2. 在step1_gen_train_test.py中的random_wrong，是将原始正确文本进行随机的单个字符替换吧？ 代码number = random.randint(672, 7992)中的672和7992是随便写的？ 我看gen_train_test函数中，train_data没进行处理，而是放在smbert_dataset.py文件中的DataFactory方法中处理，而且该处的处理方法好像也只是随机替换，并没有涉及同音字，形近字等？---也就是说char_meta.txt文件并没有用到？？
> 3. 如果只是简单的文字替换的话，那原始文本和标签文本的前后长度，还是一样的吧？而实际场景中，错误的句子，很可能是有一定的增减几率的。。而且，更多的情况下的错误会是同音字、谐音字，形近字的错误。。这样的话，模型训练出来，对上述提到的错误情况，是否有用呢？

@lmw0320 您好，请问您对问题13有解决办法吗
